scalyr_agent/tests/monitors_manager_test.py

```diff
@@ -50,10 +75,88 @@ def test_multiple_modules(self):
                 'gauss_mean': 0
             },
         ], [])
+        status = test_manager.generate_status()
+        self.assertEquals(status.total_alive_monitors, 0)
+        self.assertFalse(any([ms.is_alive for ms in status.monitors_status]))
 
-        self.assertEquals(len(test_manager.monitors), 2)
-        self.assertEquals(test_manager.monitors[0].monitor_name, 'test_monitor(1)')
-        self.assertEquals(test_manager.monitors[1].monitor_name, 'test_monitor(2)')
+        def augment_user_agent(fragments):
+            pass
+
+        test_manager.set_user_agent_augment_callback(augment_user_agent)
+        test_manager.start_manager()
+
+        status = test_manager.generate_status()
+        self.assertEquals(status.total_alive_monitors, 2)
+        self.assertTrue(all([ms.is_alive for ms in status.monitors_status]))
+
+        test_manager.stop_manager()
```

Comment:
Do you want to wait on join here?  Otherwise I think it will be flaky if those monitor threads have actually stopped or not.